libraries/AC_AutoTune/AC_AutoTune_FreqResp.cpp

```diff
@@ -281,19 +287,30 @@ void AC_AutoTune_FreqResp::push_to_tgt_buffer(uint16_t count, float amplitude, u
     sample.curr_count = count;
     sample.amplitude = amplitude;
     sample.time_ms = time_ms;
-    tgt_peak_info_buffer.push(sample);
-
+    if (tgt_peak_info_buffer != nullptr) {
+        tgt_peak_info_buffer->push(sample);
+    }
 }
 
 // pull target peak info from buffer
 void AC_AutoTune_FreqResp::pull_from_tgt_buffer(uint16_t &count, float &amplitude, uint32_t &time_ms)
 {
     peak_info sample;
-    if (!tgt_peak_info_buffer.pop(sample)) {
+    if ((tgt_peak_info_buffer == nullptr) || !tgt_peak_info_buffer->pop(sample)) {
         // no sample
         return;
     }
     count = sample.curr_count;
     amplitude = sample.amplitude;
     time_ms = sample.time_ms;
 }
+
+void AC_AutoTune_FreqResp::set_dwell_cycles(uint8_t cycles)
+{
+    dwell_cycles = cycles;
+    delete meas_peak_info_buffer;
```

Comment:
Randy or PeterB will come up with a suggestion on how we can avoid the delete by create a larger-than-required ring buffer.

Reply:
@rmackay9 @peterbarker  I removed the ring buffer delete statement and cleaned things up.  I think my code works fine without them.  I just clear the buffer before it is used the next time.  I guess the buffer pops that first thing that was given to it if it is not full?  Please have a look and let me konw.